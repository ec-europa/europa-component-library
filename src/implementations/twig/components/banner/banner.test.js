import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';

import bannerDataPrimary from '@ecl/specs-component-banner/demo/data--primary';
import bannerDataImage from '@ecl/specs-component-banner/demo/data--image-box';
import bannerDataImageShade from '@ecl/specs-component-banner/demo/data--image-shade';
import bannerDataImageGradient from '@ecl/specs-component-banner/demo/data--image-gradient';

expect.extend(toHaveNoViolations);

describe('Banner', () => {
  const template = '@ecl/banner/banner.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Primary', () => {
    const data = bannerDataPrimary;

    test(`renders correctly`, () => {
      expect.assertions(1);

      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(renderTwigFileAsHtml(template, data, true))
      ).toHaveNoViolations();
    });
  });

  describe('image', () => {
    test(`- text-box renders correctly`, () => {
      expect.assertions(1);

      return expect(render(bannerDataImage)).resolves.toMatchSnapshot();
    });

    test(`- gradient renders correctly`, () => {
      expect.assertions(1);

      return expect(render(bannerDataImageGradient)).resolves.toMatchSnapshot();
    });

    test(`- shade renders correctly`, () => {
      expect.assertions(1);

      return expect(render(bannerDataImageShade)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(bannerDataImage, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(bannerDataImage, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(renderTwigFileAsHtml(template, bannerDataImage, true))
      ).toHaveNoViolations();
    });
  });
});
