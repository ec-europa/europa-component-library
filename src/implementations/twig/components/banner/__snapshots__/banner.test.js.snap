// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Plain background renders correctly 1`] = `
<jest>
  <section
    class="ecl-banner ecl-banner--plain-background ecl-banner--m ecl-banner--centered"
    data-ecl-auto-init="Banner"
  >
    <div
      class="ecl-container"
    >
      <div
        class="ecl-banner__container"
      >
        <div
          class="ecl-banner__content"
        >
          <div
            class="ecl-banner__title"
          >
            <span
              class="ecl-banner__title-text"
            >
              Headline sed elit lorem. Donec dictum.
            </span>
          </div>
          <p
            class="ecl-banner__description"
          >
            <span
              class="ecl-banner__description-text"
            >
              Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu, dictum maximus arcu pellentesque eget. Cras massa nunc. Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu
            </span>
          </p>
          <div
            class="ecl-banner__cta"
          >
            <a
              class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
              href="/example"
            >
              <span
                class="ecl-link__label"
              >
                CTA link
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#corner-arrow"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</jest>
`;

exports[`Banner Plain background renders correctly with extra attributes 1`] = `
<jest>
  <section
    class="ecl-banner ecl-banner--plain-background ecl-banner--m ecl-banner--centered"
    data-ecl-auto-init="Banner"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <div
      class="ecl-container"
    >
      <div
        class="ecl-banner__container"
      >
        <div
          class="ecl-banner__content"
        >
          <div
            class="ecl-banner__title"
          >
            <span
              class="ecl-banner__title-text"
            >
              Headline sed elit lorem. Donec dictum.
            </span>
          </div>
          <p
            class="ecl-banner__description"
          >
            <span
              class="ecl-banner__description-text"
            >
              Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu, dictum maximus arcu pellentesque eget. Cras massa nunc. Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu
            </span>
          </p>
          <div
            class="ecl-banner__cta"
          >
            <a
              class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
              href="/example"
            >
              <span
                class="ecl-link__label"
              >
                CTA link
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#corner-arrow"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</jest>
`;

exports[`Banner Plain background renders correctly with extra class names 1`] = `
<jest>
  <section
    class="ecl-banner ecl-banner--plain-background ecl-banner--m ecl-banner--centered custom-class custom-class--test"
    data-ecl-auto-init="Banner"
  >
    <div
      class="ecl-container"
    >
      <div
        class="ecl-banner__container"
      >
        <div
          class="ecl-banner__content"
        >
          <div
            class="ecl-banner__title"
          >
            <span
              class="ecl-banner__title-text"
            >
              Headline sed elit lorem. Donec dictum.
            </span>
          </div>
          <p
            class="ecl-banner__description"
          >
            <span
              class="ecl-banner__description-text"
            >
              Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu, dictum maximus arcu pellentesque eget. Cras massa nunc. Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu
            </span>
          </p>
          <div
            class="ecl-banner__cta"
          >
            <a
              class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
              href="/example"
            >
              <span
                class="ecl-link__label"
              >
                CTA link
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#corner-arrow"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</jest>
`;

exports[`Banner image - image overlay renders correctly 1`] = `
<jest>
  <section
    class="ecl-banner ecl-banner--image-overlay ecl-banner--m ecl-banner--centered"
    data-ecl-auto-init="Banner"
  >
    <div
      class="ecl-banner__picture-container"
    >
      <picture
        class="ecl-picture ecl-banner__picture"
      >
        <img
          alt="alternative text"
          class="ecl-banner__image"
          src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
        />
      </picture>
      <div
        class="ecl-banner__credit"
      >
        © Copyright or credit
      </div>
    </div>
    <div
      class="ecl-container"
    >
      <div
        class="ecl-banner__container"
      >
        <div
          class="ecl-banner__content"
        >
          <div
            class="ecl-banner__title"
          >
            <span
              class="ecl-banner__title-text"
            >
              Headline sed elit lorem. Donec dictum.
            </span>
          </div>
          <p
            class="ecl-banner__description"
          >
            <span
              class="ecl-banner__description-text"
            >
              Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu, dictum maximus arcu pellentesque eget. Cras massa nunc.
            </span>
          </p>
          <div
            class="ecl-banner__cta"
          >
            <a
              class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
              href="/example"
            >
              <span
                class="ecl-link__label"
              >
                CTA link
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#corner-arrow"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</jest>
`;

exports[`Banner image - text highlight renders correctly 1`] = `
<jest>
  <section
    class="ecl-banner ecl-banner--text-highlight ecl-banner--m ecl-banner--centered"
    data-ecl-auto-init="Banner"
  >
    <div
      class="ecl-banner__picture-container"
    >
      <picture
        class="ecl-picture ecl-banner__picture"
      >
        <img
          alt="aternative text"
          class="ecl-banner__image"
          src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
        />
      </picture>
      <div
        class="ecl-banner__credit"
      >
        © Copyright or credit
      </div>
    </div>
    <div
      class="ecl-container"
    >
      <div
        class="ecl-banner__container"
      >
        <div
          class="ecl-banner__content"
        >
          <div
            class="ecl-banner__title"
          >
            <span
              class="ecl-banner__title-text"
            >
              Headline sed elit lorem. Donec dictum.
            </span>
          </div>
          <p
            class="ecl-banner__description"
          >
            <span
              class="ecl-banner__description-text"
            >
              Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu, dictum maximus arcu pellentesque eget. Cras massa nunc. dictum maximus arcu pellentesque eget. Cras massa nunc.
            </span>
          </p>
          <div
            class="ecl-banner__cta"
          >
            <a
              class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
              href="/example"
            >
              <span
                class="ecl-link__label"
              >
                CTA link
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#corner-arrow"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</jest>
`;

exports[`Banner image - text-box renders correctly 1`] = `
<jest>
  <section
    class="ecl-banner ecl-banner--text-box ecl-banner--m ecl-banner--centered"
    data-ecl-auto-init="Banner"
  >
    <div
      class="ecl-banner__picture-container"
    >
      <picture
        class="ecl-picture ecl-banner__picture"
      >
        <img
          alt="alternative text"
          class="ecl-banner__image"
          src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
        />
      </picture>
      <div
        class="ecl-banner__credit"
      >
        © Copyright or credit
      </div>
    </div>
    <div
      class="ecl-container"
    >
      <div
        class="ecl-banner__container"
      >
        <div
          class="ecl-banner__content"
        >
          <div
            class="ecl-banner__title"
          >
            <span
              class="ecl-banner__title-text"
            >
              Headline sed elit lorem. Donec dictum.
            </span>
          </div>
          <p
            class="ecl-banner__description"
          >
            <span
              class="ecl-banner__description-text"
            >
              Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu, dictum maximus arcu pellentesque eget. Cras massa nunc.
            </span>
          </p>
          <div
            class="ecl-banner__cta"
          >
            <a
              class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
              href="/example"
            >
              <span
                class="ecl-link__label"
              >
                CTA link
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#corner-arrow"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</jest>
`;

exports[`Banner image renders correctly with extra attributes 1`] = `
<jest>
  <section
    class="ecl-banner ecl-banner--text-box ecl-banner--m ecl-banner--centered"
    data-ecl-auto-init="Banner"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <div
      class="ecl-banner__picture-container"
    >
      <picture
        class="ecl-picture ecl-banner__picture"
      >
        <img
          alt="alternative text"
          class="ecl-banner__image"
          src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
        />
      </picture>
      <div
        class="ecl-banner__credit"
      >
        © Copyright or credit
      </div>
    </div>
    <div
      class="ecl-container"
    >
      <div
        class="ecl-banner__container"
      >
        <div
          class="ecl-banner__content"
        >
          <div
            class="ecl-banner__title"
          >
            <span
              class="ecl-banner__title-text"
            >
              Headline sed elit lorem. Donec dictum.
            </span>
          </div>
          <p
            class="ecl-banner__description"
          >
            <span
              class="ecl-banner__description-text"
            >
              Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu, dictum maximus arcu pellentesque eget. Cras massa nunc.
            </span>
          </p>
          <div
            class="ecl-banner__cta"
          >
            <a
              class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
              href="/example"
            >
              <span
                class="ecl-link__label"
              >
                CTA link
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#corner-arrow"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</jest>
`;

exports[`Banner image renders correctly with extra class names 1`] = `
<jest>
  <section
    class="ecl-banner ecl-banner--text-box ecl-banner--m ecl-banner--centered custom-class custom-class--test"
    data-ecl-auto-init="Banner"
  >
    <div
      class="ecl-banner__picture-container"
    >
      <picture
        class="ecl-picture ecl-banner__picture"
      >
        <img
          alt="alternative text"
          class="ecl-banner__image"
          src="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
        />
      </picture>
      <div
        class="ecl-banner__credit"
      >
        © Copyright or credit
      </div>
    </div>
    <div
      class="ecl-container"
    >
      <div
        class="ecl-banner__container"
      >
        <div
          class="ecl-banner__content"
        >
          <div
            class="ecl-banner__title"
          >
            <span
              class="ecl-banner__title-text"
            >
              Headline sed elit lorem. Donec dictum.
            </span>
          </div>
          <p
            class="ecl-banner__description"
          >
            <span
              class="ecl-banner__description-text"
            >
              Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu, dictum maximus arcu pellentesque eget. Cras massa nunc.
            </span>
          </p>
          <div
            class="ecl-banner__cta"
          >
            <a
              class="ecl-link ecl-link--cta ecl-link--icon ecl-banner__link-cta"
              href="/example"
            >
              <span
                class="ecl-link__label"
              >
                CTA link
              </span>
              <svg
                aria-hidden="true"
                class="ecl-icon ecl-icon--xs ecl-icon--rotate-90 ecl-link__icon"
                focusable="false"
              >
                <use
                  xlink:href="/icons.svg#corner-arrow"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</jest>
`;
